lib/services/services.c

```diff
@@ -1079,3 +1080,90 @@ resources_list_agents(const char *standard, const char *provider)
 
     return NULL;
 }
+
+gboolean
+resources_agent_exists(const char *standard, const char *provider, const char *agent)
+{
+    GList *standards = NULL;
+    GList *providers = NULL;
+    GListPtr iter = NULL;
+    gboolean rc = FALSE;
+    gboolean has_providers = FALSE;
+
+    standards = resources_list_standards();
+    for (iter = standards; iter != NULL; iter = iter->next) {
+        if (crm_str_eq(iter->data, standard, TRUE)) {
+            rc = TRUE;
+            break;
+        }
+    }
+
+    if (rc == FALSE) {
+        goto done;
+    }
+
+    rc = FALSE;
+
+    has_providers = is_set(pcmk_get_ra_caps(standard), pcmk_ra_cap_provider);
+    if (has_providers == TRUE && provider != NULL) {
+        providers = resources_list_providers(standard);
+        for (iter = providers; iter != NULL; iter = iter->next) {
+            if (crm_str_eq(iter->data, provider, TRUE)) {
+                rc = TRUE;
+                break;
+            }
+        }
+    } else if (has_providers == FALSE && provider == NULL) {
+        rc = TRUE;
+    }
+
+    if (rc == FALSE) {
+        goto done;
+    }
+
+    if (safe_str_eq(standard, PCMK_RESOURCE_CLASS_SERVICE)) {
+        if (services__lsb_agent_exists(agent)) {
+            rc = TRUE;
+#if SUPPORT_SYSTEMD
+        } else if (systemd_unit_exists(agent)) {
+            rc = TRUE;
+#endif
+
+#if SUPPORT_UPSTART
+        } else if (upstart_unit_exists(agent)) {
```

Comment:
upstart_job_exists()

FYI, our travis CI is the only platform readily available that still uses upstart, so it's good for catching these

Reply:
Yeah, I meant to go back and check travis after I pushed this to check things like that, but I got distracted looking at something else and didn't come back for it.  I'll try to remember to do that in the future.